Linting fix: gocritic assignOp, unlambda, typeSwitchVar

kensipe opened the pull request:
In the process of adding linting fixes currently working on gocritic.  This fixes:

- typeSwitchVar: which cleans up switch statement code with variable assignment 
- assignOp: which cleans up modifications to a var to not stutter
- unlamba: which removes an anonymous function which does nothing but enclose a function of the same signature
- removal of params that start with a capitalization

lizzthabet commented:
thanks for your linting prs! i'll flag them for a couple more maintainers to look at. a few folks on the tilt team are wrapping up a feature release this week, so it'll likely be after the weekend to get pr feedback / approval.

kensipe commented:
@nicks any time for a review?   I have a number of more lints to submit but didn't want to overwhelm or assume.   No worries if there are higher priorities.

nicks reviewed (commented):
thanks @kensipe ! one nit below:

re your high level question: " I have a number of more lints to submit but didn't want to overwhelm or assume."

lint fixes are welcome! they're a good way to start contributing. but also want to make sure 
(1) you're not feeling like that's the only way to contribute
(2) that we're still reviewing the lint PRs closely -- I've seen many issues where "obvious" lint PRs weren't reviewed closely enough and caused major problems (cf https://www.schneier.com/blog/archives/2008/05/random_number_b.html)

kensipe commented on internal/k8s/statefulset.go:
```diff
@@ -15,18 +15,16 @@ import (
 // Tilt should change all statefulsets to use a parallel policy.
 func InjectParallelPodManagementPolicy(entity K8sEntity) K8sEntity {
-	switch entity.Obj.(type) {
+	entity = entity.DeepCopy()
```
I'll take a look.. but I don't think I added any logic... only doing what was originally there.   checking...

kensipe replied:
but happy to fix regardless btw... just want to understand consequences

kensipe replied:
great catch @nicks !   that was definitely a mistake.  It's ready for re-review

kensipe replied:
please hold...

kensipe commented:
@nicks resolved and rebased to master... should be good.  I reverted the change to statefulset... and will have an answer on another PR with other lint findings when I can get some heads down time.  really great catch!  thanks for the review!

nicks reviewed (approved):
thanks!
